### ISSUE TITLE
Unable to play: install VLC

### ISSUE THREAD

**Issue by Elinvention** [flathub/io.webtorrent.WebTorrent#3]
I think the sandbox prevents the program from executing /usr/bin/vlc. If it's possible, how can I grant that permission?

**Comment by TingPing** [comment]
Is it configurable? You can do `flatpak run --talk-name=org.freedesktop.Flatpak io.webtorrent.WebTorrent` and then configure it to run `flatpak-spawn --host vlc`

(if that works you can make it permanent with `flatpak override --user --talk-name=org.freedesktop.Flatpak io.webtorrent.WebTorrent`)

**Comment by Elinvention** [comment]
Sorry, where can I find `flatpak-spawn`? I am using Debian testing.

**Comment by TingPing** [comment]
You don't need to find it anywhere, its what you configure inside the application to run.

**Comment by bilelmoussaoui** [comment]
The application lets you play videos that use codecs that are not supported by default within WebTorrent using VLC. You can select another executable using an open file dialog. 
I think the best way to handle this is to patch the application to use MPV instead of VLC and add MPV to the manifest if it doesn't require a lot of dependencies. Otherwise, the application is not that useful to stream torrents which is one of the best features it has...

**Comment by TingPing** [comment]
> add MPV to the manifest if it doesn't require a lot of dependencies.

https://github.com/flathub/io.github.GnomeMpv/blob/master/io.github.GnomeMpv.json#L49-L158

**Comment by bilelmoussaoui** [comment]
It seems that we can't easily patch the app to use MPV instead of VLC, instead of building the whole MPV within this one, reported upstream so they open files using xdg-open to fix this issue for us.
https://github.com/webtorrent/webtorrent-desktop/issues/1533

I suppose this can be closed as an upstream issue.

**Comment by Zorrototo** [comment]
is there a way to be able to select VLC from the file explorer which opens when you want to select the player to use? I don't really understand, what is the problem actually? from my point of view I think it is missing permissions to access the VLC executable. I'm new to Flatpak so I would take any guidance to make the appropriate changes on my side.

**Comment by TingPing** [comment]
@Zorrototo 	Flatpaks are sandboxes, they cannot see or execute anything on the host directly.

**Comment by Zorrototo** [comment]
Yep that's what I figured out after posting, that is how flatpacks work so I guess maybe a workaround could be to implement VLC with WebTorrent in the same flatpak, but that seems like complicated.
I found this to be helpful in understanding the permission http://docs.flatpak.org/en/latest/sandbox-permissions.html 

I was using this flatpack version of WebTorrent because it is currently broken on Arch based distros recently and this version here worked instantly but with these issues.

**Comment by Aex12** [comment]
> Is it configurable? You can do `flatpak run --talk-name=org.freedesktop.Flatpak io.webtorrent.WebTorrent` and then configure it to run `flatpak-spawn --host vlc`
> 
> (if that works you can make it permanent with `flatpak override --user --talk-name=org.freedesktop.Flatpak io.webtorrent.WebTorrent`)

This is the correct answer.

The webtorrent client doesnt have an option to use a custom command for the external player, instead it forces you to select a file, but you can just create a sh file and select it. Just be sure it has +x permission, and it's on the Downloads folder, so the sandbox allow it to access.

```
echo -e '#!/bin/sh\n\nflatpak-spawn --host vlc $@' > ~/Downloads/spawn-vlc.sh
chmod +x ~/Downloads/spawn-vlc.sh
flatpak override --user --talk-name=org.freedesktop.Flatpak io.webtorrent.WebTorrent
```
Then run WebTorrent, go to Edit > Preferences > Playback > External player. Click on "CHANGE" and select the file ~/Downloads/spawn-vlc.sh

It will work after that.

**Comment by S7venLights** [comment]
How would one use flatseal to allow Webtorrent access to specific file paths of video players, using the 'filesystem' permissions?
Would that even help?
Cos that .sh method is a bit of a hack that goes over my head...

**Comment by pchmykh** [comment]
What if VLC is also installed as Flatpak?
